src/pages/lessons/[slug].tsx

```diff
@@ -70,22 +72,31 @@ const VIDEO_MIN_HEIGHT = 480
 
 const Lesson: FunctionComponent<LessonProps> = ({initialLesson}) => {
   const {height} = useWindowSize()
-
+  const [lesson, setLesson] = React.useState<any>(initialLesson)
+  const {slug} = lesson
   const clientHeight = isBrowser() ? height : 0
   const [lessonMaxWidth, setLessonMaxWidth] = useState(0)
   const router = useRouter()
   const playerRef = React.useRef(null)
   const {viewer} = useViewer()
   const [playerState, send] = useMachine(playerMachine)
+  const {onProgress, onEnded} = useEggheadPlayer(lesson)
 
   const currentPlayerState = playerState.value
 
-  const lesson: any = {...initialLesson}
-
   useLastResource(lesson)
 
   const {data} = useSWR(lesson.media_url, fetcher)
 
+  React.useEffect(() => {
+    const token = getAccessTokenFromCookie()
+    loadLesson(slug, token).then((loadedLesson) => {
```

Comment:
we want to load the full lesson with the viewer's credentials and merge that into our stateful lesson